### ISSUE TITLE
Rendering in a thread

### ISSUE THREAD

**Issue by AstraLuma** [ppb/pursuedpybear#544]
Because SDL rendering is synchronous (and likely to remain so), we should look into rendering in a thread.

1. Benchmark the current renderer, see how much time is spent each frame doing: PPB math, SDL math, SDL drawing
2. Do a prototype, benchmark it

This might also let us move from fixed framerate to refresh-synced frames.

**Comment by AstraLuma** [comment]
Yeah, I pushed back on this before. My brain hadn't really processed "copying the tree" vs "generate a reduced data structure from the tree".

**Comment by AstraLuma** [comment]
Should we do this or move to a shader-based renderer?

**Comment by pathunstrom** [comment]
I think this is both, but now tying this to #661
